x/liquidity/spec/03_state_transitions.md

```diff
@@ -1,174 +1,235 @@
+## Pseudo Algorithm for LiquidityPoolBatch Execution
 
-For withdrawal, after successful withdraw, escrowed pool coins are burnt, and corresponding amount of reserve coins are sent to the withdrawer from the `Pool`.
+You can see a Python simulation script on the B-Harvest [GitHub repo](https://github.com/b-harvest/Liquidity-Module-For-the-Hub/blob/master/pseudo-batch-execution-logic/batch.py).
 
-### Pseudo Algorithm for LiquidityPoolBatch Execution
+## Swap Price Calculation
 
-simulation script (in python) : [https://github.com/b-harvest/Liquidity-Module-For-the-Hub/blob/master/pseudo-batch-execution-logic/batch.py](https://github.com/b-harvest/Liquidity-Module-For-the-Hub/blob/master/pseudo-batch-execution-logic/batch.py)
+Swap price calculations are used for these cases.
 
-**1) Swap Price Calculation**
+**Find price direction**
 
-**Finding price direction**
+Variables:
 
-- Variables
+- `X`: Reserve of X coin
+- `Y`: Reserve of Y coin before this batch execution
+- `PoolPrice` = `X`/`Y`
+- `XOverLastPrice`: amount of orders that swap X for Y with order price higher than the last `PoolPrice`
+- `XAtLastPrice`: amount of orders that swap X for Y with order price equal to the last `PoolPrice`
+- `YUnderLastPrice`: amount of orders that swap Y for X with order price lower than last `PoolPrice`
+- `YAtLastPrice`: amount of orders that swap Y for X with order price equal to the last `PoolPrice`
 
-  - `X` : Reserve of X coin, `Y` : Reserve of Y coin (before this batch execution)
-  - `PoolPrice` = `X`/`Y`
-  - `XOverLastPrice` : amount of orders which swap X for Y with order price higher than last `PoolPrice`
-  - `XAtLastPrice` : amount of orders which swap X for Y with order price equal to last `PoolPrice`
-  - `YUnderLastPrice` : amount of orders which swap Y for X with order price lower than last `PoolPrice`
-  - `YAtLastPrice` : amount of orders which swap Y for X with order price equal to last `PoolPrice`
+- **Increase**: swap price is increased from the last `PoolPrice`
 
-- **Increase** : swap price is increased from last `PoolPrice`
+  - `XOverLastPrice` > (`YUnderLastPrice`+`YAtLastPrice`)*`PoolPrice`
 
-  - `XOverLastPrice` > (`YUnderLastPrice`+`YAtLastPrice`)\*`PoolPrice`
+- **Decrease**: swap price is decreased from the last `PoolPrice`
 
-- **Decrease** : swap price is decreased from last `PoolPrice`
+  - `YUnderLastPrice` > (`XOverLastPrice`+`XAtLastPrice`)/`PoolPrice`
 
-  - `YUnderLastPrice` > (`XOverLastPrice`+`XAtLastPrice`)/`PoolPrice`
+- **Stay**: swap price is not changed from the last `PoolPrice` when the increase and decrease inequalities do not hold
 
-- **Stay** : swap price is not changed from last `PoolPrice`
-  - when both above inequalities do not hold
+### Stay case
 
-**Stay case**
+Variables:
+
+- `swapPrice` = last `PoolPrice`
+- `EX`: All executable orders that swap X for Y with order price equal to or greater than last `PoolPrice`
```

Comment:
tip: in almost all cases, we want to use "that" instead of "which" see https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which